src/util/XDRCereal.h

```diff
@@ -89,3 +89,26 @@ cereal_override(cereal::JSONOutputArchive& ar, const xdr::pointer<T>& t,
 // otherwise some interplay of name lookup and visibility
 // during the enable_if call in the cereal adaptor fails to find them.
 #include <xdrpp/cereal.h>
+
+// If name is a nonempty string, the output string begins with it.
+// If compact = true, the output string will not contain any indentation.
+template <typename T>
+std::string
+xdr_to_string(const T& t, std::string const& name = "", bool compact = false)
+{
+    std::stringstream os;
+    {
+        cereal::JSONOutputArchive ar(
```

Comment:
Probably worth a comment here about cereal flush-on-destruct RAII properties. I had to double check in the documentation about why this is required.

Reply:
Makes sense. Added a comment!